chemprop/uncertainty/evaluator.py

```diff
@@ -32,23 +34,102 @@ def evaluate(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -
 
 @UncertaintyEvaluatorRegistry.register("nll-regression")
 class NLLRegressionEvaluator(UncertaintyEvaluator):
+    r"""
+    Evaluate uncertainty values for regression datasets using the mean negative-log-likelihood
+    of the targets given the probability distributions estimated by the model:
+
+    .. math::
+
+        \mathrm{NLL}(y, \hat y) = \frac{1}{2} \log(2 \pi \sigma^2) + \frac{(y - \hat{y})^2}{2 \sigma^2},
+
+    where :math:`\hat{y}` is the predicted value, :math:`y` is the true value, and
+    :math:`\sigma^2` is the predicted uncertainty (variance).
+
+    The function returns a tensor containing the mean NLL for each task.
+    """
+
     def evaluate(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -> Tensor:
-        ...
-        return
+        nlls = []
+        for j in range(uncs.shape[1]):
+            mask_j = mask[:, j]
+            preds_j = preds[:, j]
+            targets_j = targets[:, j]
+            uncs_j = uncs[:, j]
+            masked_preds = preds_j[mask_j]
+            masked_targets = targets_j[mask_j]
+            masked_uncs = uncs_j[mask_j]
+            errors = masked_preds - masked_targets
+            nll = (2 * torch.pi * masked_uncs).log() / 2 + errors**2 / (2 * masked_uncs)
+            nlls.append(nll.mean(dim=0))
+        return torch.stack(nlls)
 
 
 @UncertaintyEvaluatorRegistry.register("nll-classification")
 class NLLClassEvaluator(UncertaintyEvaluator):
+    """
+    Evaluate uncertainty values for binary classification datasets using the mean negative-log-likelihood
+    of the targets given the assigned probabilities from the model:
+
+    .. math::
+
+        \mathrm{NLL} = -\log(\sigma \cdot y + (1 - \sigma) \cdot (1 - y))
+
+    where :math:`y` is the true binary label (0 or 1), and
+    :math:`\sigma` is the predicted probability associated with the class label `1`.
```

Comment:
Should we use $\hat{y}\$ instead of $\sigma$?

Reply:
I have changed it to use $\hat{y}$.